pkg/cloud/gcp/actuators/machine/reconciler_test.go

```diff
@@ -330,6 +330,62 @@ func TestCreate(t *testing.T) {
 				}
 			},
 		},
+		{
+			name: "Restart policy defaults to Always when not set",
+			validateInstance: func(t *testing.T, instance *compute.Instance) {
+				if instance.Scheduling != nil && instance.Scheduling.AutomaticRestart != nil && *instance.Scheduling.AutomaticRestart != true {
+					t.Errorf("Expected to find instance automatic restart set to true")
```

Comment:
nit: the `name` is stale; it should be "...to nil when not set...".

And the final `*instance.Scheduling.AutomaticRestart != true` seems to allow `AutomaticRestart == nil` and `AutomaticRestart == &true`, while we only want to allow the former.  Can we use:

```go
if instance.Scheduling != nil && instance.Scheduling.AutomaticRestart != nil {
```

for "it's fine if you have no `Scheduling` opinions.  Or if you have scheduling opinions but no `AutomaticRestart` opinions"?

Reply:
i think i'm going to refactor these tests so that we are just testing the functionality of the new code i am adding. this is starting to feel like we are testing the gcp sdk and i don't think that is appropriate since we don't control the sdk.